### ISSUE TITLE
[PS] Permissions page: Consider (conditionally) removing the checkboxes for the administrator role

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6187]
This is another idea that although won't completely eradicate the issue, will likely minimize the chances of hitting the `max_input_vars ` limit on the "Permissions" admin page (see issues like #3824, #5907 etc.): So if the following two conditions are true:
1. the role has been granted all possible/available permissions already
2. the "Default administrator role" setting in the "Default roles" section in `admin/config/people/roles` is set to the administrator role (which means that any new permission is to be granted to that role automatically)

...then we could:
- either not render the column for that role in the permissions matrix at all - throw a message at the top of the page instead (or something like that)
- or remove all the actual, non-dummy checkboxes and leave only the dummy ones in their place for that column (the dummy checkboxes are NOT included in POST - see #5907)
 
If a site has either way too many permissions, or their `max_input_vars` limit set to a very low value (or a combination of both of these conditions), then chances are that the site would still have this issue. However, we would reduce the chances of this happening in many cases, so that'd be a win.

Noting that the two conditions that I have mentioned above are both true for new installations (in the `standard` installation profile, which [seems to be what 99% of Backdrop site are using](https://backdropcms.org/project/backdrop/telemetry)), and the benefit of this proposal here would be "transparent" for any sites that never touch the default settings in `admin/config/people/roles` and always like their default, OOTB admin role to have all permissions. This would be a huge win for smaller sites that have more chances to be hosted in less powerful servers (with php limits set to very low values) and that gradually install/enable many modules that provide new permissions.

Thoughts?

**Comment by yorkshire-pudding** [comment]
I was working on a contrib module just this week [Enforce Revision Logs](https://github.com/enforce_revlog) where actually the permissions are about overriding settings, which is plausible, based on my experience, and also of recent events on `b.org` where, even though an admin could theoretically go and grant themselves the permission, you may want to disable the permission for admins and even user1 to ensure revisions happen with a log message every time on the basis that someone is unlikely to bother changing the permission and will create a revision and log even if the log is keyboard mash or a single word.  This module may be an edge case, but how would we enable admin permissions to be disabled?

Filter Permissions has a role filter, as does Role Help which can do this if needed; if this change happened, would it break such modules?

**Comment by kiamlaluno** [comment]
/admin/config/people/permissions still allows to change the permissions for the role that has been set as *Default administrator role*. IMO, those permissions should still be editable, whether they are all selected.

**Comment by klonos** [comment]
@yorkshire-pudding what I am proposing here is not to alter how the permissions system works nor how permissions are being overridden - the idea is that that would remain the same.

@kiamlaluno hmm 🤔 yes ...that does make it tricky. I can still think of a way to still optimize things in a similar manner, but without compromising the editability of the permissions. Let me think about it before I elaborate on that, as it would reduce the scope (end potential benefit) considerably, so might not even be worth the effort.

**Comment by kiamlaluno** [comment]
@klonos To make it clear what I mean: If the role set as default administrator role is like the user whose ID is 1 and has all the permissions, I would not show any permission for that role  or show the permissions that role has without making possible to change them. In The last case, that could be done without using input elements, which would reduce the submitted values.

**Comment by kiamlaluno** [comment]
Indeed, in the case the permissions for the default administrator role can be changed, the permission checkboxes must be rendered even when all the checkboxes are initially all selected.

**Comment by yorkshire-pudding** [comment]
@klonos / @kiamlaluno - I was giving an example of a (possibly niche) use case when you might want to disable permissions from admin role to help with not being able to override something. Another use case is where you are developing a member directory where as an admin, you want to limit what you can see on a normal basis.

I'm not saying it won't work; but there needs to be a way to disable permissions for admin. Just because someone starts with all permissions doesn't mean they won't want to be removed.

With Filter Permissions, it is collapsed unless a filter is applied with it or `max-input-vars` is low. Maybe, some approach along those lines is needed?

**Comment by bugfolder** [comment]
> the description for the Default administrator role setting says _This role will be automatically given any permission introduced by newly enabled modules_. It does not say those permissions are kept.

Furthermore, that doesn't actually always happen. CiviCRM, in particular, introduces new permissions in updates or extensions that show up as unchecked for the default administrator role. (In fact, I just went and checked a recently Civi-updated site, and found a few more.) So, hiding checkboxes for the default administrator role would make it harder for administrators to discover permissions that were unchecked.

**Comment by kiamlaluno** [comment]
@bugfolder My bad: I got confused with Drupal 8, where the *Administrator* role always has all the permissions, new or old permissions.  
Even Drupal 10 shows the *Administrator* permissions on /admin/people/permissions, but they cannot be edited and are all selected.

**Comment by klonos** [comment]
Thanks everyone for your input here 🙏🏼 

> Even Drupal 10 shows the *Administrator* permissions on /admin/people/permissions, but they cannot be edited and they are all selected.

Yes @kiamlaluno, that difference with D8/9/10 always trips me up too. Sometimes I find that easier/better, other times I appreciate the flexibility in Backdrop to revoke certain permissions from the default admin role. One way to remember how things work is that basically the admin role in modern Drupal basically behaves (or is treated) the same as user1.

Still thinking about whether this request here should be adapted or the issue be closed as "works as designed" 🤔
